plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -198,7 +198,7 @@ public void createNamespace(ConnectorSession session, String namespace, Map<Stri
                 case LOCATION_PROPERTY -> {
                     String location = (String) value;
                     try {
-                        fileSystemFactory.create(session).newInputFile(Location.of(location)).exists();
+                        fileSystemFactory.create(session).directoryExists(Location.of(location));
```

Comment:
i don't understand how the optional empty is supposed to work.

in particular, while s3 has no concept of normal directories, it has a concept of file listings and hive, iceberg and delta connectors use it effectively in the same manner as directories. (hive uses that for "all" operations, while others mostly for cleanups & drop)

there are places where we want to prevent creation of a new empty thing (partition, table, schema) with a location pointing at somewhere where files can already be found, so e.g. prevent two tables from sharing a directory location.

I don't see how `directoryExists` API, returning empty on S3, so basically disabled there, can be used to deliver the safety checks that we want

Reply:
cc @alexjo2144 @pajaks @findinpath @losipiuk @ebyhr @krvikash

Reply:
The `directoryExists()` method is explicitly for directories and only directories, intended for usage in places where the existence of a possibly empty directory matters. S3 doesn't have directories. If you want to know if a location has files, then you need to list files.

`directoryExists()` was added for the Hive `checkPartitionLocationExists()` check, which needs to fail if the directory doesn't exist, but succeed if there are no files (i.e. empty directory). The check always succeeds on S3.

If it turns out that we need another API that combines the two, we'll add it or change this one as appropriate.

Reply:
Maybe we need a new API, this I am not judging now.
The move to current `directoryExists` is a wrong though, since it's not the API we want here.

Also, s3 has directories-by-convention, so from user perspective there is a concept of empty directories. Not sure why would we want to ignore that aspect.

Reply:
Why is this the wrong API here? What is the exact behavior that you intend to have happen here?

Reply:
> s3 has directories-by-convention, so from user perspective there is a concept of empty directories. Not sure why would we want to ignore that aspect

If you're referring to the fake zero-length objects that are directories, those are rare, and not used by Trino or other data lake systems (to my knowledge).  For example, `TrinoS3FileSystem.mkdirs()` is a no-op.

Reply:
I updated the contract as described above. I believe this will address your concerns.